hexagonal/src/main/java/com/iluwatar/hexagonal/domain/LotteryNumbers.java

```diff
@@ -85,11 +84,10 @@ public static LotteryNumbers create(Set<Integer> givenNumbers) {
    * @return numbers as comma separated string
    */
   public String getNumbersAsString() {
-    List<Integer> list = new ArrayList<>();
-    list.addAll(numbers);
     StringBuilder builder = new StringBuilder();
+    Iterator<Integer> iterator = numbers.iterator();
     for (int i = 0; i < NUM_NUMBERS; i++) {
-      builder.append(list.get(i));
+      builder.append(iterator.next());
```

Comment:
Why we don't have a check for `iterator.hasNext()`

Reply:
The same reason we didn't check a list boundaries.